src/tree.js

```diff
@@ -83,16 +84,23 @@ export function createInternal(vnode, parentInternal) {
 		props,
 		key,
 		ref,
+		data: null,
+		rerender: null,
+		flags,
 		_children: null,
 		_parent: parentInternal,
 		_vnodeId: vnodeId,
 		_dom: null,
 		_component: null,
-		flags,
 		_context: null,
 		_depth: parentInternal ? parentInternal._depth + 1 : 0
 	};
 
+	if (typeof type == 'function') {
+		internal.rerender = enqueueRender.bind(null, internal);
+		internal.data = {};
```

Comment:
Could compute this the same way the other properties are computed differently for function vs element Internals (on L70):

```diff
function createInternal() {
+  let data;
	// ...
+		if (flags & TYPE_ELEMENT === 0) {
+			data = {};
+		} else if (type === 'svg') {
-		if (flags & TYPE_ELEMENT && type === 'svg') {
	// ...  

	const internal = {
		// ...
+		data,
+		render: enqueueRender,
	}
}
```